cargo/src/cargo_commands.rs

```diff
@@ -0,0 +1,342 @@
+    } else {
+        cargo_update(curdir, &first_manifest.to_string_lossy())?;
+    }
+    cargo_fetch(curdir, &first_manifest.to_string_lossy(), update)?;
+    cargo_generate_lockfile(curdir, &first_manifest.to_string_lossy(), update)?;
+
+    // NOTE: Vendor filterer's default output format is directory so we
+    // don't need to set that ourselves.
+    let res = cargo_command(which_subcommand, &default_options, curdir);
+
+    if possible_lockfile.is_file() {
+        default_options.push("--locked".to_string());
+        info!(?possible_lockfile, "🔓 Adding lockfile.");
+        lockfiles.push(possible_lockfile.as_path().to_path_buf());
+        let bytes = fs::read(&possible_lockfile)?;
+        hasher2.update(&bytes);
+    }
+    let hash1 = hex::encode(hasher1.finalize());
+    let hash2 = hex::encode(hasher2.finalize());
+    if hash1 != hash2 {
+        debug!(?hash1, ?hash2);
+        warn!("⚠️ Lockfile has changed");
+        warn!("Previous hash: {}", hash1);
+        warn!("New hash: {}", hash2);
+        warn!(
+			"⚠️ If you wish to respect the lockfile, consider not setting `--update` to true. However, \
+			 this MIGHT FAIL in some cases."
+		);
+        if has_update_value_changed && update {
+            let mut msg: String = "⚠️ Update was SET from FALSE to TRUE , hence a NEW LOCKFILE was CREATED since there was \
+        NO LOCKFILE prior. Your dependencies MIGHT have updated.".to_string();
+            if filter {
+                msg.push_str(" This is because `--filter` option is set to true 🧺.");
+            }
+            warn!(msg);
+        }
+    } else {
+        info!(
+            "🔒 Lockfile was not regenerated for `{}`",
+            possible_lockfile.display()
+        );
+        info!("Previous hash: {}", hash1);
+        info!("New hash: {}", hash2);
+    }
+
+    info!("🛡️🫥 Auditing lockfiles...");
+    if let Ok(audit_result) = audit::perform_cargo_audit(&lockfiles, i_accept_the_risk) {
+        audit::process_reports(audit_result).map_err(|err| {
+            error!(?err);
+            io::Error::new(io::ErrorKind::Interrupted, err.to_string())
+        })?;
+    }
+    info!("🛡️🙂 All lockfiles are audited");
+    match res {
+        Ok(output_cargo_configuration) => {
+            info!("🏪 `cargo {}` finished.", &which_subcommand);
+            Ok(Some((
+                possible_lockfile
+                    .canonicalize()
+                    .unwrap_or(possible_lockfile),
+                output_cargo_configuration,
+            )))
+        }
+        Err(err) => {
+            error!(?err);
+            Err(err)
+        }
+    }
+}
+
+pub fn cargo_generate_lockfile(
+    curdir: &Path,
+    manifest: &str,
+    mut update: bool,
+) -> io::Result<String> {
+    info!("🔓 💂 Running `cargo generate-lockfile`...");
+    let mut has_update_value_changed = false;
+    let mut hasher1 = Keccak256::default();
```

Comment:
I think if you just want fast verification, sha2 is plenty (and it's cpu accelerated!)

Reply:
okii me will test sha2 out.

Reply:
If we're talking about speed there is always [BLAKE3](https://github.com/BLAKE3-team/BLAKE3) which claims it is way faster than SHA-2 or SHA-3